aten/CMakeLists.txt

```diff
@@ -386,16 +376,23 @@ else()
   add_subdirectory(src/THCS)
 endif()
 
+find_package(CuDNN)
+IF(NOT AT_CUDA_ENABLED OR NOT CUDNN_FOUND)
+  MESSAGE(STATUS "CuDNN not found. Compiling without CUDNN support")
+  set(AT_CUDNN_ENABLED 0)
+ELSE()
+  INCLUDE_DIRECTORIES(${CUDNN_INCLUDE_DIRS})
+  set(AT_CUDNN_ENABLED 1)
+ENDIF()
+
 set(cwrap_files
   ${CMAKE_CURRENT_SOURCE_DIR}/src/ATen/Declarations.cwrap
   ${CMAKE_CURRENT_SOURCE_DIR}/src/THNN/generic/THNN.h
   ${CMAKE_CURRENT_SOURCE_DIR}/src/THCUNN/generic/THCUNN.h
   ${CMAKE_CURRENT_SOURCE_DIR}/src/ATen/nn.yaml
   ${CMAKE_CURRENT_SOURCE_DIR}/src/ATen/native/native_functions.yaml
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/ATen/native/cudnn/native_functions.yaml
```

Comment:
Why do we need to define these in a different file?  In the spirit of your ATen philosophy note, I'd argue that these are like any other ATen native function, although they may have different implementation details (i.e. they throw exceptions if you don't have cudnn defined).  It seems to me we want all native functions to be declared in the same place (whether that is a single .yaml file or a bunch of .yaml files in the same directory broken up functionally).  It's certainly okay for the implementations to be defined separately (as you've done).